# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/WebrtcGlobalInformation.h
# Commit: fe152aff2a22
# Full Hash: fe152aff2a22e3336dd4696538cbb26779c42fce
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2022-12-22 17:00:32
# Regressor Bug: 1800136
# File Overlap Count: 1
# Description:
#   Bug 1800136: Fix gap between PC.close() and the storage of its final stats in PeerConnectionCtx. r=ng
#   
#   Ensures that:
#   1) We never have stats report in a still-registered PC _and_ the stats for
#      closed PCs.
# ==============================================================================

diff -r 99ea4f221834 -r fe152aff2a22 dom/media/webrtc/jsapi/WebrtcGlobalInformation.h
--- a/dom/media/webrtc/jsapi/WebrtcGlobalInformation.h	Thu Dec 22 02:33:57 2022 +0000
+++ b/dom/media/webrtc/jsapi/WebrtcGlobalInformation.h	Thu Dec 22 02:33:57 2022 +0000
@@ -17,6 +17,7 @@
 class GlobalObject;
 class WebrtcGlobalStatisticsCallback;
 class WebrtcGlobalLoggingCallback;
+struct RTCStatsReportInternal;
 
 class WebrtcGlobalInformation {
  public:
@@ -42,7 +43,7 @@
   static bool AecDebug(const GlobalObject& aGlobal);
   static void GetAecDebugLogDir(const GlobalObject& aGlobal, nsAString& aDir);
 
-  static void StoreLongTermICEStatistics(PeerConnectionImpl& aPc);
+  static void StashStats(const RTCStatsReportInternal& aReport);
 
  private:
   WebrtcGlobalInformation() = delete;
